acceptance/scripts/generic/testrun.sh

```diff
@@ -14,18 +13,60 @@ export BEAKER_HELPER="acceptance/lib/helper.rb"
 
 bundle install --path vendor/bundle
 
-bundle exec genconfig2 $GENCONFIG_LAYOUT > $BEAKER_CONFIG
-
-bundle exec beaker \
-  --config $BEAKER_CONFIG \
-  --type aio \
-  --helper $BEAKER_HELPER \
-  --options-file $BEAKER_OPTIONS \
-  --tests $BEAKER_TESTSUITE \
-  --post-suite $BEAKER_POSTSUITE \
-  --pre-suite $BEAKER_PRESUITE \
-  --keyfile $BEAKER_KEYFILE \
-  --load-path "acceptance/lib" \
-  --preserve-hosts onfail \
+#TODO: Someday, when our Rakefile is refactored, this section will go away
+BEAKER="bundle exec beaker --debug"
+BEAKER="$BEAKER --type aio"
+BEAKER="$BEAKER --keyfile $BEAKER_KEYFILE"
+BEAKER="$BEAKER --helper $BEAKER_HELPER"
+BEAKER="$BEAKER --options-file $BEAKER_OPTIONS"
+BEAKER="$BEAKER --post-suite $BEAKER_POSTSUITE"
+BEAKER="$BEAKER --load-path acceptance/lib"
+
+
+case $1 in
+  -p | --p* )
+  bundle exec genconfig2 $GENCONFIG_LAYOUT > $BEAKER_CONFIG
+    
+  if [ -z "$PACKAGE_BUILD_VERSION" ]; 
+    #TODO: curl builds.puppetlabs.lan/puppetserver.  Parse HTML, find most recent folder name.
+    then echo "$0: PACKAGE_BUILD_VERSION not set.  Please export PACKAGE_BUILD_VERSION."
+    exit -1;
+  fi
+  
+  BEAKER="$BEAKER --pre-suite $BEAKER_PRESUITE"
+  BEAKER="$BEAKER --tests $BEAKER_TESTSUITE"
+  BEAKER="$BEAKER --config $BEAKER_CONFIG"
+  BEAKER="$BEAKER --preserve-hosts"
+  $BEAKER
+  echo "Beaker exited with $?"
+  cp log/latest/hosts_preserved.yml .
+  ;;
+
+
+-r | --r* )
+  if [ ! -s ./hosts_preserved.yml ];
+  then echo "$0: Can not find hosts_preserved.yml; can not run without presuite./n \
```

Comment:
My Ruby stinks, but there seems to be extra or missing characters before the line-continuation: ```...presuite./n \```. Is that a typo or does my Ruby really stink?